### PR TITLE
ApplyPatchAction: Keep track of origs

### PR DESCRIPTION
**Pull request by sils1297**
There's no need to crawl *every directory* every time for origs. This is
probably a huge performance hit on HDDs.

### CODE REVIEW

File: tests/coalaCITest.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 import re
 import sys
 import unittest
-from tempfile import NamedTemporaryFile, TemporaryDirectory
 
 from coalib import coala_ci
-from coalib.misc.ContextManagers import make_temp, prepare_file
+from coalib.misc.ContextManagers import make_temp
+from coalib.misc.ContextManagers import prepare_file
```
Comment (hypothesist) [inline tests/coalaCITest.py]: Unrelated change?

Comment (hypothesist) [inline tests/coalaCITest.py]: Actually, you have to remove them

Comment (hypothesist) [inline tests/coalaCITest.py]: Remove only `make_temp`

**Comment by hypothesist** [comment]
ack https://github.com/coala-analyzer/coala/pull/2320/commits/c911fd24b61bff7f26929352b9ca8f30a180aa11

**Comment by hypothesist** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge
